src/datachain/lib/signal_schema.py

```diff
@@ -384,6 +392,37 @@ def deserialize(schema: dict[str, Any]) -> "SignalSchema":
 
         return SignalSchema(signals)
 
+    @staticmethod
```

Comment:
Q: why is it a static method?

Q: should we rename it? get_flatten_hidden_fields or something like that ...

Reply:
It is static method because it doesn't really use any values from the class and makes it easy to call from Studio.

Reply:
Renamed to get_flatten_hidden_fields.

Reply:
I mean, should be just schema.get_flatten_hidden_fields() instead (instance method). It's a bit strange to see that we class method and pass pretty much an instance of the same class

Reply:
I see the confusion, we are not passing the schema object itself, but we are passing the serialized part of feature schema we have at the moment inside the dataset version. We don't need to form a signalschema object just for this IMO.

Reply:
Ok, I see. Parameter should be renamed then? So, it means we kinda expose some internals (operate on the serialized format). What if that format changes? It means we'll have to have some backward compatibility? Is it compatible with version 1? (before things like bases were introduced)? are we testing this per format?

Reply:
It is from dataset versions we save in database.  It does have backward compatibility because we access the field like below if you check the PR changes:
```
schema.get("hidden_fields", [])
```
and so on.

Reply:
but it means we'll have to always now keep in mind in one extra place this compatibility aspect? can we avoid using serialized format? (I think @dreadatour is kinda looking into a similar issue btw)

Reply:
I don't think so, I mean we already are using the feature schema in other places directly in studio and probably in datachain. I think having to create signal schema object just to parse the certain fields within sounds expensive to me specially when listing the versions.

Reply:
We maintain the format in CustomType object in itself in the same file for this.

Reply:
> and probably in datachain

hopefully not (?)

> in other places directly in studio

means it should be improved if we see more place coming (like in this case)?

>  I think having to create signal schema object just to parse the certain fields within sounds expensive to me specially when listing the versions.

it's a single `deserialize` call on a smallish dict unless I'm missing something? what does deserialize do, does it create models and such?

Reply:
> it's a single deserialize call on a smallish dict unless I'm missing something? what does deserialize do, does it create models and such?

Deserialization creates a signal based on the field. The issue is the deserialized signal doesn't have the type definition with it so, we did not have the hidden signal with us. I tried implementing it in  e6767be997ed09822e5851a99b996002b075b1f9 but it did not work in studio.  We need to look into the feature schema as we are already doing in Studio already.

Reply:
I honestly do not see the benefit in calling a class method by initializing a class instead of a static class.  TBH